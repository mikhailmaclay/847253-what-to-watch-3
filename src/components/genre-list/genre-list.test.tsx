// Libraries
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// Components
import GenreList from './genre-list';
//
import emptyArrowFunction from '../../utils/components/empty-arrow-function';

describe(`<GenreList/>`, () => {
  const props = {
    genres: [`Comedy`],
    currentGenre: null,
    changeGenre: emptyArrowFunction
  };

  it(`should render correctly`, () => {
    const result = renderer.create(
        <GenreList {...props}/>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
