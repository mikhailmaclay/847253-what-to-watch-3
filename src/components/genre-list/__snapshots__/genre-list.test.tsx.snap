// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreList/> should render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <button
      className="catalog__genres-link"
      disabled={true}
      onClick={null}
      style={
        Object {
          "background": "transparent",
          "border": "none",
          "cursor": "",
          "paddingLeft": "0",
          "paddingRight": "0",
          "userSelect": "none",
        }
      }
      type="button"
    >
      All genres
    </button>
  </li>
  <li
    className="catalog__genres-item "
  >
    <button
      className="catalog__genres-link"
      disabled={false}
      href="#"
      onClick={[Function]}
      style={
        Object {
          "background": "transparent",
          "border": "none",
          "cursor": "pointer",
          "paddingLeft": "0",
          "paddingRight": "0",
          "userSelect": "none",
        }
      }
      type="button"
    >
      Comedy
    </button>
  </li>
</ul>
`;
